itemframework/src/item/abstract_item.cpp

```diff
@@ -325,6 +327,38 @@ ItemOutput* AbstractItem::addOutput(int type, QString const& description)
     return output;
 }
 
+void AbstractItem::removeInput(ItemInput* itemInput)
```

Comment:
I'm not sure why I made it in a loop, to find an object that I already know. 

`void AbstractItem::removeOutput(ItemOutput *itemOutput)
{
    Q_D(AbstractItem);
    
    if(itemOutput != nullptr){
        itemOutput->disconnectInputs();
        d->_outputs.removeOne(itemOutput);
        d->realignOutputs();
        delete itemOutput;
        update();
    }
}`